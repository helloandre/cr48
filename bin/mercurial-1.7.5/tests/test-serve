#!/bin/sh

hgserve()
{
    hg serve -a localhost -d --pid-file=hg.pid -E errors.log -v $@ \
        | sed -e "s/:$HGPORT1\\([^0-9]\\)/:HGPORT1\1/g" \
              -e "s/:$HGPORT2\\([^0-9]\\)/:HGPORT2\1/g" \
              -e 's/http:\/\/[^/]*\//http:\/\/localhost\//'
    cat hg.pid >> "$DAEMON_PIDS"
    echo % errors
    cat errors.log
    sleep 1
    if [ "$KILLQUIETLY" = "Y" ]; then
        kill `cat hg.pid` 2>/dev/null
    else
        kill `cat hg.pid`
    fi
    sleep 1
}

hg init test
cd test

echo '[web]' > .hg/hgrc
echo 'accesslog = access.log' >> .hg/hgrc
echo "port = $HGPORT1" >> .hg/hgrc

echo % Without -v
hg serve -a localhost -p $HGPORT -d --pid-file=hg.pid -E errors.log
cat hg.pid >> "$DAEMON_PIDS"
if [ -f access.log ]; then
    echo 'access log created - .hg/hgrc respected'
fi
echo % errors
cat errors.log

echo % With -v
hgserve

echo % With -v and -p HGPORT2
hgserve -p "$HGPORT2"

echo '% With -v and -p daytime (should fail because low port)'
KILLQUIETLY=Y
hgserve -p daytime
KILLQUIETLY=N

echo % With --prefix foo
hgserve --prefix foo

echo % With --prefix /foo
hgserve --prefix /foo

echo % With --prefix foo/
hgserve --prefix foo/

echo % With --prefix /foo/
hgserve --prefix /foo/
