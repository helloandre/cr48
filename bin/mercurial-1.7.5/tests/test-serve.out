% Without -v
access log created - .hg/hgrc respected
% errors
% With -v
listening at http://localhost/ (bound to 127.0.0.1:HGPORT1)
% errors
% With -v and -p HGPORT2
listening at http://localhost/ (bound to 127.0.0.1:HGPORT2)
% errors
% With -v and -p daytime (should fail because low port)
abort: cannot start server at 'localhost:13': Permission denied
abort: child process failed to start
% errors
% With --prefix foo
listening at http://localhost/foo/ (bound to 127.0.0.1:HGPORT1)
% errors
% With --prefix /foo
listening at http://localhost/foo/ (bound to 127.0.0.1:HGPORT1)
% errors
% With --prefix foo/
listening at http://localhost/foo/ (bound to 127.0.0.1:HGPORT1)
% errors
% With --prefix /foo/
listening at http://localhost/foo/ (bound to 127.0.0.1:HGPORT1)
% errors
